src/lib.rs

```diff
@@ -190,6 +190,41 @@
 //!     claims.email().map(|email| email.as_str()).unwrap_or("<not provided>"),
 //! );
 //!
+//! // If available, we can use the UserInfo endpoint to request additional information
+//!
+//! // First we need a struct that contains the AdditionalClaims we want to retrieve
+//! #[derive(Clone, Debug, Deserialize, PartialEq, Serialize)]
+//! struct MyClaims {
+//!     roles: String,
+//! }
+//! // We use an empty implementation
+//! impl AdditionalClaims for MyClaims {}
+//!
+//! // Now we create a new user_info request using the client
+//! let userinfo_request = match client.user_info(
+//! AccessToken::new(
+//!   // We re-use the access_token we used for requesting the IdToken
+//!   token_response.access_token().secret().to_string()),
+//!   None,
+//! ) {
+//!   Ok(o) => o,
```

Comment:
let's replace this match and the one below with the more common `.map_err(|err| ...)?` pattern

Reply:
How about now?